File: android/app/src/main/java/org/fossasia/openevent/config/strategies/LanguageStrategy.java
package org.fossasia.openevent.config.strategies;

import android.content.Context;

import org.fossasia.openevent.config.ConfigStrategy;

import java.util.Locale;

import lombok.Getter;
import lombok.Setter;

/**
 * Sets and provides default system language
 * To be used via {@link org.fossasia.openevent.config.StrategyRegistry}
 */
@Setter
@Getter
public class LanguageStrategy implements ConfigStrategy {

    private String defaultSystemLanguage;

    @Override
    public boolean configure(Context context) {
        defaultSystemLanguage = Locale.getDefault().getDisplayLanguage();
        return false;
    }

}

File: android/app/src/main/java/org/fossasia/openevent/config/strategies/LanguageStrategy.kt
package org.fossasia.openevent.config.strategies

import android.content.Context
import org.fossasia.openevent.config.ConfigStrategy
import java.util.*

/**
 * Sets and provides default system language
 * To be used via [org.fossasia.openevent.config.StrategyRegistry]
 */
class LanguageStrategy : ConfigStrategy {

    var defaultSystemLanguage: String? = null

    override fun configure(context: Context): Boolean {
        defaultSystemLanguage = Locale.getDefault().displayLanguage
        return false
    }

}
